### PR TITLE
Sessions external tools

### PR DESCRIPTION
**Pull request by kavinvalli**

### CODE REVIEW

**Comment by kavinvalli** [comment]
@chitalian This should be ready but I feel like we can discuss a bit more on the type of request especially for the vector db call, looks kind of raw to me right now

File: helicone-node/index2.ts
Comment (chitalian) [inline helicone-node/index2.ts:1]: index2 is a suspicious

File: valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts [hunk 1]
```diff
@@ -41,6 +41,19 @@ export class RequestBodyHandler extends AbstractLogHandler {
       ) {
         context.processedLog.request.model = "assistant-call";
       }
+
+      console.log("requestBody type", requestBodyFinal._type);
```
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts:45]: rm please

File: valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts [hunk 2]
```diff
@@ -41,6 +41,19 @@ export class RequestBodyHandler extends AbstractLogHandler {
...
+
+      if (this.isVectorDBRequest(requestBodyFinal)) {
+        console.log("is vector db");
+        context.processedLog.request.model = "vector_db";
+      }
+
+      if (this.isToolRequest(requestBodyFinal)) {
```
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts:52]:
```suggestion
      } else if (this.isToolRequest(requestBodyFinal)) {
```

File: valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts [hunk 3]
```diff
@@ -41,6 +41,19 @@ export class RequestBodyHandler extends AbstractLogHandler {
...
+        console.log("is tool", requestBodyFinal.toolName);
```
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/RequestBodyHandler.ts:53]: rm

File: valhalla/jawn/src/lib/handlers/ResponseBodyHandler.ts [hunk 4]
```diff
@@ -248,6 +260,15 @@ export class ResponseBodyHandler extends AbstractLogHandler {
       return { responseModel: "Assistant Polling", model: "assistant-polling" };
     } else if (this.isAssistantResponse(responseBody) && !currentModel) {
       return { responseModel: "Assistant Call", model: "assistant-call" };
+    } else if (this.isVectorDBResponse(responseBody)) {
+      console.log("response is vector db");
```
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/ResponseBodyHandler.ts:264]: rm

File: valhalla/jawn/src/lib/handlers/ResponseBodyHandler.ts [hunk 5]
```diff
@@ -248,6 +260,15 @@ export class ResponseBodyHandler extends AbstractLogHandler {
...
+      return { responseModel: "Vector DB", model: "vector_db" };
+    } else if (this.isToolResponse(responseBody)) {
+      console.log("response is tool", responseBody.toolName);
```
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/ResponseBodyHandler.ts:267]: rm

File: helicone-helpers/manual_logger/HeliconeManualLogger.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    return this.results;
+  }
+}
+
+/*
+ * Type Definitions
+ *
+ * */
+
+type ProviderRequest = {
```
Comment (chitalian) [inline helicone-helpers/manual_logger/HeliconeManualLogger.ts:121]: Can you move these to a types.ts tfile please?

File: helicone-helpers/manual_logger/HeliconeManualLogger.ts [hunk 7]
```diff
@@ -0,0 +1,178 @@
+export class HeliconeManualLogger {
+  private apiKey: string;
+  private headers: Record<string, string>;
+  private readonly LOGGING_ENDPOINT: string =
```
Comment (chitalian) [inline helicone-helpers/manual_logger/HeliconeManualLogger.ts:4]: would be great to have this configurable but defaulted on the constructor

Comment (kavinvalli) [reply helicone-helpers/manual_logger/HeliconeManualLogger.ts:6]: Hey @chitalian , I didn't understand this one - isn't this being done
https://github.com/Helicone/helicone/blob/ce8085630010a05acc64907b8b9d70798994a935/helicone-helpers/manual_logger/HeliconeManualLogger.ts#L9-L12

Comment (chitalian) [reply helicone-helpers/manual_logger/HeliconeManualLogger.ts:12]: specifically the logging endpoint

**Pull request merged**